contrib/mesos/pkg/queue/delay.go

```diff
@@ -164,6 +188,7 @@ func (q *DelayQueue) pop(next func() *qitem, cancel <-chan struct{}) interface{}
 			select {
 			case <-cancel:
 				item.readd(item)
+				sitAndSpin(&q.cond, ch)
```

Comment:
prevents an (unlikely) goroutine leak (any future broadcast received by the Wait would cause the above goroutine to exit)